src/lib/components/workspace/WorkspaceContent.svelte

```diff
@@ -17,30 +15,44 @@
   } from "$lib/utils/interfaces";
   import TeamInvitePopup from "./team-invite-popup/TeamInvitePopup.svelte";
   import { base64ToURL } from "$lib/utils/helpers";
+  import type { TeamDocument } from "$lib/database/app.database";
+  import { TeamViewModel } from "../../../pages/Teams/team.viewModel";
+  import type { Observable } from "rxjs";
+  import { PeopleIcon, ShowMoreIcon } from "$lib/assets/app.asset";
+  import { CustomButton, IconButton } from "$lib/components";
+
+  export let userId: string;
   export let data: any;
-  export let loaderColor = "default",
-    userId: string,
-    handleWorkspaceSwitch: any,
-    handleWorkspaceTab: any,
-    workspaceMethods: WorkspaceMethods,
-    activeSideBarTabMethods: any,
-    openTeam,
-    currentTeam: CurrentTeam,
-    handleCreateWorkspace: any,
+  export let loaderColor = "default";
+  export let handleWorkspaceSwitch: any;
+  export let handleWorkspaceTab: any;
+  export let activeSideBarTabMethods: any;
+  export let openTeam;
+  export let currentTeam: CurrentTeam;
+  export let handleCreateWorkspace: any,
     teamServiceMethods: TeamServiceMethods,
     teamRepositoryMethods: TeamRepositoryMethods,
     workspaces;
-  let currOpenedTeam: CurrentTeam;
-  const openedTeamSubscribe = openedTeam.subscribe((value) => {
-    if (value) currOpenedTeam = value;
-  });
+  export let handleLeaveTeamModal: () => void;
+  export let handleOnShowMoreClick: (e) => void;
+  export let isShowMoreVisible: boolean = false;
 
+  let currOpenedTeam: CurrentTeam;
   let isLoading: boolean = false;
-
   let selectedTab = "all-workspace";
   let selectedView: string;
+  let currOpenedTeamRxDoc: Observable<TeamDocument>;
+  let isTooltipVisible: boolean = false;
 
-  let selectedViewSubscribe = workspaceView.subscribe((value) => {
+  const _teamViewModel = new TeamViewModel();
+

```

Comment:
TeamViewModel Object is already initiated at parent component, you can pass your methods wrapping in TeamServiceMethods or TeamRepositoryMethods as a props.